sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/PodInstance.java

```diff
@@ -22,6 +22,10 @@ default String getName() {
     }
 
     static String getName(PodSpec podSpec, int index) {
-        return String.format("%s-%d", podSpec.getType(), index);
+        return getName(podSpec.getType(), index);
+    }
+
+    public static String getName(String podType, int index) {
```

Comment:
What case might have this being called with something other than `podSpec.getType()`? This might be a small instance of YAGNI.